### PR TITLE
Add track candidate

### PR DESCRIPTION
**Pull request by Rob1in**

### CODE REVIEW

**Review by bhaney — APPROVED** [review]
The smallest requests to just delete some lines, and then an answer to a question

File: src/tracker/track.py [hunk 1]
```diff
@@ -37,6 +48,12 @@ def __eq__(self, other) -> bool:
             and np.array_equal(self.feature_vector, other.feature_vector)
         )
 
+    def promote_track(self):
+        if not self.is_candidate:
+            raise ValueError("track has already been promoted")
```
[the remark this answers is not in the document]
Comment (Rob1in) [reply src/tracker/track.py:53]: I am not actually using that functions, forgot to delete it. 
`promote_to_track()` is a function of the `Tracker` class.

**Comment by Rob1in** [comment]
Made the changes

File: src/tracker/tracker.py [hunk 2]
```diff
@@ -193,53 +207,110 @@ def update(self, img, visualize: bool = False):
...
-                feature_dist = self.encoder.compute_distance(
-                    track.feature_vector, feature_vectors[j]
-                )
-                # Cost function: lambda * feature distance + (1 - lambda) * (1 - IoU)
-                cost_matrix[i, j] = self.lambda_value * feature_dist + (
-                    1 - self.lambda_value
-                ) * (1 - iou_score)
         # Solve the linear assignment problem to find the best matching
-        row_indices, col_indices = linear_sum_assignment(cost_matrix)
+        # row_indices, col_indices = linear_sum_assignment(cost_matrix)
```
[the remark this answers is not in the document]
Comment (Rob1in) [reply src/tracker/tracker.py]: done

File: src/tracker/encoder.py [hunk 3]
```diff
@@ -98,7 +100,8 @@ def compute_features(self, image: np.ndarray, detections: List[Detection]):
         # features = self.extractor(processed_images)
 
         features = self.extractor(cropped_images)
-        return features
+        # return features
```
[the remark this answers is not in the document]
Comment (Rob1in) [reply src/tracker/encoder.py]: done

**Pull request merged**